multiplayer/src/types/index.ts

```diff
@@ -46,6 +48,19 @@ export namespace Cli2Srv {
         type: "start-game";
     };
 
+    export type InputMessage = MessageBase & {
+        type: "input";
+        data: {
```

Comment:
`data` scoping seems unnecessary here.

Reply:
discussed offline: I had left it as is to avoid requiring a backend change tied up with this, we'll need to version messages eventually & can swap over with that change.